query-planner-js/src/buildPlan.ts

```diff
@@ -565,7 +558,10 @@ class QueryPlanningTraversal<RV extends Vertex> {
     debug.log(() => `Query has ${planCount} possible plans`);
 
     let firstBranch = this.closedBranches[0];
-    while (planCount > MAX_COMPUTED_PLANS && firstBranch.length > 1) {
+    // Note: typing is not able to know that `maxEvaluatedPlans` is guaranteed to be set here, but it is
+    // due to early call to `enforceQueryPlannerConfigDefaults`.
+    const maxPlansToCompute = this.parameters.config.debug.maxEvaluatedPlans!;
```

Comment:
I really hate this. What if you created a new type and used it in place of `Concrete<QueryPlannerConfig>` in places it's used.

```
export type ConcreteQueryPlannerConfig = {
  [Property in keyof QueryPlannerConfig]-?: Concrete<QueryPlannerConfig[Property]>
}
```

That will let the properties of `debug` also be concrete.

Reply:
Good point, didn't realised I could just call `Concrete` "recursively" on the property types. I did that in the definition of `Concrete` directly though, instead of creating a new specific type.

Reply:
Oh nice. I didn't realize you could do that either.